dev/benchmarks/complex_layout/test/scroll_smooth_measure.dart

```diff
@@ -0,0 +1,206 @@
+  binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.benchmarkLive;
+  binding.reportData ??= <String, dynamic>{};
+  final ResampleFlagVariant variant = ResampleFlagVariant(binding);
+  testWidgets('Smoothness test', (WidgetTester tester) async {
+    app.main();
+    await tester.pumpAndSettle();
+    final Finder scrollerFinder = find.byKey(const ValueKey<String>('complex-scroll'));
+    final ListView scroller = tester.widget<ListView>(scrollerFinder);
+    final ScrollController controller = scroller.controller;
+    final List<int> frameTimestamp = <int>[];
+    final List<double> scrollOffset = <double>[];
+    binding.addPersistentFrameCallback((Duration timeStamp) {
+      if (controller.hasClients) {
+        // This if is necessary because by the end of the test the widget tree
+        // is destroyed.
+        frameTimestamp.add(timeStamp.inMicroseconds);
+        scrollOffset.add(controller.offset);
+      }
+    });
+
+    Duration now() => binding.currentSystemFrameTimeStamp;
+    Future<List<Duration>> scroll() async {
+      // Extra 50ms to avoid timeouts.
+      final Duration startTime = const Duration(milliseconds: 500) + now();
+      final List<Duration> delays = <Duration>[];
+      for (final PointerDataRecord record in dragInputDatas(startTime, tester.getCenter(scrollerFinder))) {
+        await tester.binding.delayed(record.timeStamp - now());
+        // This now measures how accurate the above delayed is.
+        delays.add(now() - record.timeStamp);
+        ui.window.onPointerDataPacket(record.data);
+      }
+      return delays;
+    }
+
+    List<Duration> delays = <Duration>[];
+    for (int n = 0; n < 5; n++) {
+      delays += await scroll();
+    }
+    variant.result = scrollSummary(scrollOffset, delays, frameTimestamp);
+    await tester.pumpAndSettle();
+    scrollOffset.clear();
+    delays.clear();
+    await tester.idle();
+  }, semanticsEnabled: false, variant: variant);
+}
+
+Map<String, dynamic> scrollSummary(
+  List<double> scrollOffset,
+  List<Duration> delays,
+  List<int> frameTimestamp,
+) {
+  double jankyCount = 0;
+  double jerkAvg = 0;
+  int lostFrame = 0;
+  for (int i = 1; i < scrollOffset.length-1; i += 1) {
+    if (frameTimestamp[i+1] - frameTimestamp[i-1] > 40E3
+        || delays[i] > const Duration(milliseconds: 16)) {
+      // filter datapoints from slow frame building or input simulation artifact
+      lostFrame += 1;
+      continue;
+    }
+    // Using abs rather than square because square (2-norm) amplifies the
+    // effect of the data point that's relatively large, but in this metric
+    // we prefer smaller data point to have similiar effect.
+    // This is also why we count the number of data that's larger than a
+    // threshold (and the result is tested not sensitive to this threshold),
+    // which is effectly a 0-norm.
+    final double jerk = (scrollOffset[i-1] + scrollOffset[i+1] - 2*scrollOffset[i]).abs();
+    jerkAvg += jerk;
```

Comment:
Also document that the jerk here is not `$ d^3 x / dt^3 $ as typically defined in physics. Your explanation in the PR description should be helpful here.

Ideally, the `jerk` and `jank` should be documented in the signature of this function as these are the return of this function.

Reply:
Done.